mathics/core/symbols.py

```diff
@@ -456,6 +334,42 @@ def format_expr(expr):
         finally:
             evaluation.dec_recursion_depth()
 
+    def equal2(self, rhs: Any) -> Optional[bool]:
```

Comment:
As far as I can tell `SameQ` and `Equal` are different, both operators are pervasive over all kinds of expression types. And the note that the specific implementation here is not the same as SameQ. 

That said, possibly it could be attached in some other class and just be made something that is mandatory to define. 

Your thoughts @mmatera ?

Reply:
``SameQ`` is something that all elements should handle.  ``equal2`` I do not think so, since in general require  an  ``Evaluation`` object to be determined.